packages/react-native/src/ReactNativeFileSystem.ts

```diff
@@ -4,17 +4,26 @@ import { TypedArrayEncoder, AriesFrameworkError, getDirFromFilePath, Buffer } fr
 import * as RNFS from 'react-native-fs'
 
 export class ReactNativeFileSystem implements FileSystem {
-  public readonly basePath
+  public readonly dataPath
+  public readonly cachePath
+  public readonly tempPath
 
   /**
    * Create new ReactNativeFileSystem class instance.
    *
-   * @param basePath The base path to use for reading and writing files. RNFS.TemporaryDirectoryPath if not specified
+   * @param baseDataPath The base path to use for reading and writing data files used within the framework.
+   * Files will be created under baseDataPath/.afj directory. If not specified, it will be set to homedir()
+   * @param baseCachePath The base path to use for reading and writing cache files used within the framework.
+   * Files will be created under baseCachePath/.afj directory. If not specified, it will be set to homedir()
+   * @param baseTempPath The base path to use for reading and writing temporary files within the framework.
+   * Files will be created under baseTempPath/.afj directory. If not specified, it will be set to tmpdir()
    *
    * @see https://github.com/itinance/react-native-fs#constants
    */
-  public constructor(basePath?: string) {
-    this.basePath = basePath ?? RNFS.TemporaryDirectoryPath
+  public constructor(options?: { baseDataPath?: string; baseCachePath?: string; baseTempPath?: string }) {
+    this.dataPath = `${options?.baseDataPath ?? RNFS.DocumentDirectoryPath}/.afj`
+    this.cachePath = `${options?.baseCachePath ?? RNFS.CachesDirectoryPath}/.afj`
+    this.tempPath = `${options?.baseTempPath ?? RNFS.TemporaryDirectoryPath}/.afj`
```

Comment:
On Android, the TemporaryDirectoryPath will redirect to the Caching Directory, which means you should be aware you could have conflicts between cache & temp as they end up in the same space on Android. 

> The absolute path to the temporary directory (falls back to Caching-Directory on Android)

Reply:
Good point! Added a temp/cache suffix when platform is android